packages/x-components/src/components/result/base-result-image.vue

```diff
@@ -41,170 +41,199 @@
-    }
+    },
+    props: {
+      /**
+       * (Required) The {@link @empathyco/x-types#Result | result} information.
+       *
+       * @public
+       */
+      result: {
+        type: Object as PropType<Result>,
+        required: true
+      },
+
+      /**
+       * Animation to use when switching between the placeholder, the loaded image, or the failed
+       * image fallback.
+       *
+       * @public
+       */
+      loadAnimation: {
+        type: animationProp,
+        default: () => NoElement
+      },
+
+      /**
+       * Animation to use when switching between the loaded image and the hover image.
+       *
+       * @public
+       */
+      hoverAnimation: {
+        type: animationProp
+      },
+
+      /**
+       * Indicates if the next valid image should be displayed on hover.
+       *
+       * @public
+       */
+      showNextImageOnHover: {
+        type: Boolean,
+        default: false
+      }
+    },
 
-    /**
-     * Gets the src from the result image.
-     *
-     * @returns The result image src.
-     *
-     * @internal
-     */
-    protected get imageSrc(): string {
-      return this.loadedImages[
-        !this.showNextImageOnHover || !this.isHovering ? 0 : this.loadedImages.length - 1
-      ];
-    }
+    setup(props) {
+      /**
+       * Copy of the images of the result.
+       *
+       * It is used as a queue of images to load, once an image loads/fails to load, it is removed
+       * from this array.
+       *
+       * @internal
+       */
+      const pendingImages: Ref<string[]> = ref([]);
+
+      /**
+       * Contains the images that have been loaded successfully.
+       *
+       * @internal
+       */
+      const loadedImages: Ref<string[]> = ref([]);
+
+      /**
+       * Indicates if the user is hovering the image.
+       *
+       * @internal
+       */
+      const isHovering = ref(false);
+
+      /**
+       * Indicates if the user has hovered the image.
+       *
+       * @internal
+       */
+      const userHasHoveredImage = ref(false);
+
+      /**.
+       * Styles to use inline in the image loader, to prevent override from CSS
+       *
+       * @internal
+       */
+      const loaderStyles: Partial<CSSStyleDeclaration> = {
+        position: 'absolute !important',
+        top: '0 !important',
+        left: '0 !important',
+        width: '100% !important',
+        height: '100% !important',
+        pointerEvents: 'none !important',
+        visibility: 'hidden !important'
+      };
 
-    /**
-     * Indicates if the loader should try to load the next image.
-     *
-     * @returns True if it should try to load the next image.
-     *
-     * @internal
-     */
-    protected get shouldLoadNextImage(): boolean {
-      const numImagesToLoad = this.showNextImageOnHover && this.userHasHoveredImage ? 2 : 1;
-      return !!this.pendingImages.length && this.loadedImages.length < numImagesToLoad;
-    }
+      const resultImages = ref(props.result.images!);
```

Comment:
This breaks the reactivity of the images, use this instead:

```suggestion
      const resultImages = toRef(props.result, 'images');
```